solidity/contracts/isms/multisig/AbstractWeightedMultisigIsm.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+// ============ External Imports ============
+import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import {OwnableUpgradeable} from "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+// ============ Internal Imports ============
+import {IInterchainSecurityModule} from "../../interfaces/IInterchainSecurityModule.sol";
+import {IStaticWeightedMultisigIsm} from "../../interfaces/isms/IWeightedMultisigIsm.sol";
+import {Message} from "../../libs/Message.sol";
+
+import {MerkleLib} from "../../libs/Merkle.sol";
+import {AbstractMultisig} from "./AbstractMultisigIsm.sol";
+
+/**
+ * @title AbstractStaticWeightedMultisigIsm
+ * @notice Manages per-domain m-of-n Validator sets with stake weights that are used to verify
+ * interchain messages.
+ */
+abstract contract AbstractStaticWeightedMultisigIsm is
+    AbstractMultisig,
+    IStaticWeightedMultisigIsm
+{
+    // ============ Constants ============
+    uint96 public constant BASIS_POINTS = 10000;
```

Comment:
im not sure what this means

Reply:
oh now I see that you are treating this as "percentages"
I would personally just use absolute stake amounts as weights for simplicity

Reply:
Then we'll need to keep updating the threshold weight.

Reply:
renamed to TOTAL_WEIGHT and using 1e10 instead.

Reply:
> Then we'll need to keep updating the threshold weight.
I still view the "updates" as relatively infrequent (on an epoch/monthly basis)